src/components/select/select.ts

```diff
@@ -455,7 +455,7 @@ export class Select extends BaseInput<any> implements OnDestroy {
           return isCheckedProperty(selectValue, option.value);
         });
 
-        if (option.selected) {
+        if (option.selected && option.value) {
```

Comment:
I think we should implement a better way to check the existence of option.value.
what if the value=0? it would not work